aviatrix/resource_aviatrix_datadog_agent.go

```diff
@@ -0,0 +1,114 @@
+	return &schema.Resource{
+		Create: resourceAviatrixDatadogAgentCreate,
+		Read:   resourceAviatrixDatadogAgentRead,
+		Update: resourceAviatrixDatadogAgentUpdate,
+		Delete: resourceAviatrixDatadogAgentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"api_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "API key",
+			},
+			"site": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "datadoghq.com",
+				ValidateFunc: validation.StringInSlice([]string{"datadoghq.com", "datadoghq.eu"}, false),
+				Description:  "Site preference",
+			},
+			"excluded_gateways": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of excluded gateways.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Enabled or not.",
+			},
+		},
+	}
+}
+
+func marshalDatadogAgentInput(d *schema.ResourceData) *goaviatrix.DatadogAgent {
+	return &goaviatrix.DatadogAgent{
+		ApiKey: d.Get("api_key").(string),
+		Site:   d.Get("site").(string),
+	}
+}
+
+func resourceAviatrixDatadogAgentCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	datadogAgent := marshalDatadogAgentInput(d)
+
+	var excludedGateways []string
+	for _, v := range d.Get("excluded_gateways").(*schema.Set).List() {
+		excludedGateways = append(excludedGateways, v.(string))
+	}
+	if len(excludedGateways) != 0 {
+		datadogAgent.ExcludedGatewaysInput = strings.Join(excludedGateways, ",")
+	}
+
+	if err := client.EnableDatadogAgent(datadogAgent); err != nil {
+		return fmt.Errorf("could not enable datadog agent: %v", err)
+	}
+
+	d.SetId("datadog_agent")
+	return nil
+}
+func resourceAviatrixDatadogAgentRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	if d.Id() != "datadog_agent" {
+		return fmt.Errorf("invalid ID, expected ID \"datadog_agent\", instead got %s", d.Id())
+	}
+
+	datadogAgentStatus, err := client.GetDatadogAgentStatus()
+	if err != nil {
+		return fmt.Errorf("could not get remote syslog status: %v", err)
+	}
+
+	d.Set("api_key", datadogAgentStatus.ApiKey)
+	d.Set("site", datadogAgentStatus.Site)
+	d.Set("excluded_gateways", datadogAgentStatus.ExcludedGateways)
+	d.Set("status", datadogAgentStatus.Status)
+
+	d.SetId("datadog_agent")
+	return nil
+}
+
+func resourceAviatrixDatadogAgentUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceAviatrixDatadogAgentCreate(d, meta)
+}
+
+func resourceAviatrixDatadogAgentDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	if err := client.DisableDatadogAgent(); err != nil {
+		return fmt.Errorf("could not disable datadog agent: %v", err)
+	}
+
+	d.SetId("")
```

Comment:
I don't think this is necessary. If the Delete function returns with a `nil` error then the resource  is removed from state, no need to set an empty ID